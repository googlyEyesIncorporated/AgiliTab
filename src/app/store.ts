import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import itemListReducer from "../features/itemList/itemListSlice";
import settingsSlice from "../features/Settings/settingsSlice";

export const store = configureStore({
  reducer: {
    itemList: itemListReducer,
    settings: settingsSlice,
  },
  devTools: true,
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
