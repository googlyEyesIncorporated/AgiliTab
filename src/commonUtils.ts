export const DATE_TIME_NO_SECONDS = "yyyy-LL-dd'T'HH:mm";
