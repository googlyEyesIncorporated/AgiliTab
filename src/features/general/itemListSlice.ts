import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import { RootState } from "../../app/store";
import {
  ItemList,
  ItemListState,
  JustListKey,
  ListAndIndex,
  ListAndItem,
  ReplaceList,
} from "./types";
import { updateStorage } from "./storageHelper";

let shortTermList: ItemList = [];
let mediumTermList: ItemList = [];
let longTermList: ItemList = [];

const initialTodoListState: ItemListState = {
  shortTermList,
  mediumTermList,
  longTermList,
};

export const itemListSlice = createSlice({
  name: "todo",
  initialState: initialTodoListState,
  // The `reducers` field lets us define reducers and generate associated actions
  reducers: {
    add: (state, action: PayloadAction<ListAndItem>) => {
      const { listKey, item } = action.payload;
      state[listKey].push(item);
      updateStorage({ storageKey: listKey, val: state[listKey] });
    },
    remove: (state, action: PayloadAction<ListAndIndex>) => {
      const { listKey, index } = action.payload;
      state[listKey].splice(index, 1);
      updateStorage({ storageKey: listKey, val: state[listKey] });
    },
    toggleChecked: (state, action: PayloadAction<ListAndIndex>) => {
      const { listKey, index } = action.payload;
      state[listKey][index].done = !state[listKey][index].done;
      updateStorage({ storageKey: listKey, val: state[listKey] });
    },
    clearDone: (state, action: PayloadAction<JustListKey>) => {
      const { listKey } = action.payload;
      state[listKey] = state[listKey].filter((item) => {
        return !item.done;
      });
      updateStorage({ storageKey: listKey, val: state[listKey] });
    },
    clearAll: (state, action: PayloadAction<JustListKey>) => {
      const { listKey } = action.payload;
      state[listKey] = [];
      updateStorage({ storageKey: listKey, val: state[listKey] });
    },
    updateList: (state, action: PayloadAction<ReplaceList>) => {
      const { listKey, itemList, save = true } = action.payload;
      state[listKey] = itemList;
      save && updateStorage({ storageKey: listKey, val: state[listKey] });
    },
    populateTasksFromChrome: (
      state,
      { payload }: PayloadAction<ItemListState | {}>
    ) => {
      if (
        "shortTermList" in payload &&
        "mediumTermList" in payload &&
        "longTermList" in payload
      ) {
        const { shortTermList, mediumTermList, longTermList } = payload;
        state.shortTermList = shortTermList;
        state.mediumTermList = mediumTermList;
        state.longTermList = longTermList;
      }
    },
  },
});

export const {
  add,
  remove,
  toggleChecked,
  clearAll,
  clearDone,
  updateList,
  populateTasksFromChrome,
} = itemListSlice.actions;

// The function below is called a selector and allows us to select a value from
// the state. Selectors can also be defined inline where they're used instead of
// in the slice file. For example: `useSelector((state: RootState) => state.counter.value)`
export const selectAllLists = (state: RootState) => state.itemList;
export const selectShortTermList = (state: RootState) =>
  state.itemList.shortTermList;
export const selectMediumTermList = (state: RootState) =>
  state.itemList.mediumTermList;
export const selectLongTermList = (state: RootState) =>
  state.itemList.longTermList;

export default itemListSlice.reducer;
