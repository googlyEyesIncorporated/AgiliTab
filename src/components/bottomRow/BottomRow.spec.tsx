import { fireEvent, render, screen } from "@testing-library/react";
import { BottomRow } from ".";
import { Provider } from "react-redux";
import { store } from "../../app/store";
import "@testing-library/jest-dom";
import { addTerm } from "../../features/itemList/itemListSlice";
import { DateTime } from "luxon";
import { DATE_TIME_NO_SECONDS } from "../../commonUtils";
import userEvent from "@testing-library/user-event";
import { act } from "react-dom/test-utils";

const WrappedBottomRow = (
  <Provider store={{ ...store }}>
    <BottomRow />
  </Provider>
);

const user = userEvent.setup();
const Now = DateTime.now();

describe("bottomRow", () => {
  it("should display more than one list", () => {
    const { rerender } = render(WrappedBottomRow);
    const list1 = {
      listKey: "0",
      listObject: {
        type: "date" as const,
        startDate: Now.startOf("day").toFormat(DATE_TIME_NO_SECONDS),
        endDate: Now.endOf("day").toFormat(DATE_TIME_NO_SECONDS),
        list: [],
        title: "Today",
      },
    };
    const list2 = {
      listKey: "1",
      listObject: {
        type: "date" as const,
        startDate: Now.startOf("month").toFormat(DATE_TIME_NO_SECONDS),
        endDate: Now.endOf("month").toFormat(DATE_TIME_NO_SECONDS),
        list: [],
        title: "Month",
      },
    };
    act(() => {
      store.dispatch(addTerm(list1));
      store.dispatch(addTerm(list2));
    });
    rerender(WrappedBottomRow);
    const bottomRow = screen.queryByTestId("bottom-row");
    expect(bottomRow).toBeInTheDocument();
    expect(bottomRow).toMatchSnapshot();
  });

  it("should update the list title when title is edited", async () => {
    const { rerender } = render(WrappedBottomRow);
    const list3 = {
      listKey: "2",
      listObject: {
        type: "date" as const,
        startDate: Now.startOf("year").toFormat(DATE_TIME_NO_SECONDS),
        endDate: Now.endOf("year").toFormat(DATE_TIME_NO_SECONDS),
        list: [],
        title: "Year",
      },
    };
    const replacementText = "Replacement Text";
    act(() => {
      store.dispatch(addTerm(list3));
    });
    rerender(WrappedBottomRow);
    fireEvent.click(
      screen.getByTestId("group-0-settings").querySelector("svg") as Element
    );
    const titleInput = screen.getByTestId("group-0-unit-name");
    user.clear(titleInput);
    user.click(titleInput);
    user.paste(replacementText);
    rerender(WrappedBottomRow);
    expect(screen.getByText(replacementText)).toBeInTheDocument();
  });

  it("should clear all items when clicking clear all", () => {
    const { rerender } = render(WrappedBottomRow);

    const listWithItems = {
      listKey: "3",
      listObject: {
        type: "date" as const,
        startDate: Now.startOf("year").toFormat(DATE_TIME_NO_SECONDS),
        endDate: Now.endOf("year").toFormat(DATE_TIME_NO_SECONDS),
        list: [
          { id: "1", name: "item 1", done: true },
          { id: "2", name: "item 2", done: false },
        ],
        title: "Year",
      },
    };
    act(() => {
      store.dispatch(addTerm(listWithItems));
    });
    rerender(WrappedBottomRow);
    expect(
      screen.queryByText(listWithItems.listObject.list[0].name)
    ).toBeInTheDocument();
    expect(
      screen.queryByText(listWithItems.listObject.list[1].name)
    ).toBeInTheDocument();
    fireEvent.click(screen.getByTestId("group-3-clear-all-button"));
    rerender(WrappedBottomRow);
    expect(
      screen.queryByText(listWithItems.listObject.list[0].name)
    ).not.toBeInTheDocument();
    expect(
      screen.queryByText(listWithItems.listObject.list[1].name)
    ).not.toBeInTheDocument();
  });

  it("should clear done items when clicking clear done", () => {
    const { rerender } = render(WrappedBottomRow);
    const listWithItems = {
      listKey: "4",
      listObject: {
        type: "date" as const,
        startDate: Now.startOf("year").toFormat(DATE_TIME_NO_SECONDS),
        endDate: Now.endOf("year").toFormat(DATE_TIME_NO_SECONDS),
        list: [
          { id: "1", name: "item 1", done: true },
          { id: "2", name: "item 2", done: false },
        ],
        title: "Year",
      },
    };
    act(() => {
      store.dispatch(addTerm(listWithItems));
    });
    rerender(WrappedBottomRow);
    expect(
      screen.queryByText(listWithItems.listObject.list[0].name)
    ).toBeInTheDocument();
    expect(screen.queryByText("item 2")).toBeInTheDocument();

    fireEvent.click(screen.getByTestId("group-4-clear-done-button"));
    rerender(WrappedBottomRow);
    expect(
      screen.queryByText(listWithItems.listObject.list[0].name)
    ).not.toBeInTheDocument();
    expect(screen.queryByText("item 2")).toBeInTheDocument();
  });

  it("should cross out the item when checked", () => {
    // setup
    const { rerender } = render(WrappedBottomRow);

    const listWithItems = {
      listKey: "5",
      listObject: {
        type: "date" as const,
        startDate: Now.startOf("year").toFormat(DATE_TIME_NO_SECONDS),
        endDate: Now.endOf("year").toFormat(DATE_TIME_NO_SECONDS),
        list: [{ id: "1", name: "item 1", done: false }],
        title: "Year",
      },
    };
    act(() => {
      store.dispatch(addTerm(listWithItems));
    });
    rerender(WrappedBottomRow);

    // click checkbox
    fireEvent.click(
      screen
        .getByTestId("group-5-list")
        .querySelector('[data-testid="list-item-checkbox-0"]') as Element
    );
    rerender(WrappedBottomRow);
    expect(
      screen
        .getByTestId("group-5-list")
        .querySelector('[data-testid="todo-text-0"]')
    ).toHaveClass("line-through");
  });

  it("should update the list item name when edited", async () => {
    // setup
    render(WrappedBottomRow);
    const newTaskName = "Take out the garbage";
    const oldTaskName = "Task 1";

    const listWithItems = {
      listKey: "6",
      listObject: {
        type: "date" as const,
        startDate: Now.startOf("year").toFormat(DATE_TIME_NO_SECONDS),
        endDate: Now.endOf("year").toFormat(DATE_TIME_NO_SECONDS),
        list: [{ id: "1", name: oldTaskName, done: false }],
        title: "Year",
      },
    };
    act(() => {
      store.dispatch(addTerm(listWithItems));
    });

    // Edit list item
    fireEvent.dblClick(
      screen.getByTestId("group-6-list").querySelector("li") as Element
    );
    fireEvent.change(screen.getByTestId("todo-edit-0"), {
      target: { value: newTaskName },
    });
    fireEvent.click(screen.getByTestId("group-6-list"));

    expect(screen.getByText(newTaskName)).toBeInTheDocument();
  });
});
