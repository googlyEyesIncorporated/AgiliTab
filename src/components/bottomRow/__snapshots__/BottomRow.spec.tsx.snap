// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bottomRow should display 3 lists, shortTerm, mediumTerm, longTerm 1`] = `
<div
  class="flex justify-between lg:h-1/2 lg:flex-nowrap h-auto flex-wrap"
  data-testid="bottom-row"
>
  <div
    class="fade-in-up-1s align-top m-2 w-full lg:w-3/10 inline-block"
    data-testid="group-container"
  >
    <div
      class="pb-1 text-2xl border-current text-center border-b"
      data-testid="group-0-header"
      role="columnheader"
      tabindex="0"
    >
      <div
        class="absolute left-0"
      >
        <div
          class="inline-block"
          data-testid="group-0-settings"
        >
          <span
            aria-hidden="true"
            class="cursor-pointer mr-2 hidden"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-gears "
              data-icon="gears"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M308.5 135.3c7.1-6.3 9.9-16.2 6.2-25c-2.3-5.3-4.8-10.5-7.6-15.5L304 89.4c-3-5-6.3-9.9-9.8-14.6c-5.7-7.6-15.7-10.1-24.7-7.1l-28.2 9.3c-10.7-8.8-23-16-36.2-20.9L199 27.1c-1.9-9.3-9.1-16.7-18.5-17.8C173.9 8.4 167.2 8 160.4 8h-.7c-6.8 0-13.5 .4-20.1 1.2c-9.4 1.1-16.6 8.6-18.5 17.8L115 56.1c-13.3 5-25.5 12.1-36.2 20.9L50.5 67.8c-9-3-19-.5-24.7 7.1c-3.5 4.7-6.8 9.6-9.9 14.6l-3 5.3c-2.8 5-5.3 10.2-7.6 15.6c-3.7 8.7-.9 18.6 6.2 25l22.2 19.8C32.6 161.9 32 168.9 32 176s.6 14.1 1.7 20.9L11.5 216.7c-7.1 6.3-9.9 16.2-6.2 25c2.3 5.3 4.8 10.5 7.6 15.6l3 5.2c3 5.1 6.3 9.9 9.9 14.6c5.7 7.6 15.7 10.1 24.7 7.1l28.2-9.3c10.7 8.8 23 16 36.2 20.9l6.1 29.1c1.9 9.3 9.1 16.7 18.5 17.8c6.7 .8 13.5 1.2 20.4 1.2s13.7-.4 20.4-1.2c9.4-1.1 16.6-8.6 18.5-17.8l6.1-29.1c13.3-5 25.5-12.1 36.2-20.9l28.2 9.3c9 3 19 .5 24.7-7.1c3.5-4.7 6.8-9.5 9.8-14.6l3.1-5.4c2.8-5 5.3-10.2 7.6-15.5c3.7-8.7 .9-18.6-6.2-25l-22.2-19.8c1.1-6.8 1.7-13.8 1.7-20.9s-.6-14.1-1.7-20.9l22.2-19.8zM112 176a48 48 0 1 1 96 0 48 48 0 1 1 -96 0zM504.7 500.5c6.3 7.1 16.2 9.9 25 6.2c5.3-2.3 10.5-4.8 15.5-7.6l5.4-3.1c5-3 9.9-6.3 14.6-9.8c7.6-5.7 10.1-15.7 7.1-24.7l-9.3-28.2c8.8-10.7 16-23 20.9-36.2l29.1-6.1c9.3-1.9 16.7-9.1 17.8-18.5c.8-6.7 1.2-13.5 1.2-20.4s-.4-13.7-1.2-20.4c-1.1-9.4-8.6-16.6-17.8-18.5L583.9 307c-5-13.3-12.1-25.5-20.9-36.2l9.3-28.2c3-9 .5-19-7.1-24.7c-4.7-3.5-9.6-6.8-14.6-9.9l-5.3-3c-5-2.8-10.2-5.3-15.6-7.6c-8.7-3.7-18.6-.9-25 6.2l-19.8 22.2c-6.8-1.1-13.8-1.7-20.9-1.7s-14.1 .6-20.9 1.7l-19.8-22.2c-6.3-7.1-16.2-9.9-25-6.2c-5.3 2.3-10.5 4.8-15.6 7.6l-5.2 3c-5.1 3-9.9 6.3-14.6 9.9c-7.6 5.7-10.1 15.7-7.1 24.7l9.3 28.2c-8.8 10.7-16 23-20.9 36.2L315.1 313c-9.3 1.9-16.7 9.1-17.8 18.5c-.8 6.7-1.2 13.5-1.2 20.4s.4 13.7 1.2 20.4c1.1 9.4 8.6 16.6 17.8 18.5l29.1 6.1c5 13.3 12.1 25.5 20.9 36.2l-9.3 28.2c-3 9-.5 19 7.1 24.7c4.7 3.5 9.5 6.8 14.6 9.8l5.4 3.1c5 2.8 10.2 5.3 15.5 7.6c8.7 3.7 18.6 .9 25-6.2l19.8-22.2c6.8 1.1 13.8 1.7 20.9 1.7s14.1-.6 20.9-1.7l19.8 22.2zM464 304a48 48 0 1 1 0 96 48 48 0 1 1 0-96z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
        <div
          class="inline-block"
          data-testid="group-0-copy"
        >
          <span
            aria-hidden="true"
            class="cursor-pointer mr-2 hidden"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-copy "
              data-icon="copy"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M208 0H332.1c12.7 0 24.9 5.1 33.9 14.1l67.9 67.9c9 9 14.1 21.2 14.1 33.9V336c0 26.5-21.5 48-48 48H208c-26.5 0-48-21.5-48-48V48c0-26.5 21.5-48 48-48zM48 128h80v64H64V448H256V416h64v48c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V176c0-26.5 21.5-48 48-48z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </div>
      <span
        class="text-[1.6875rem]"
        data-testid="group-0-title"
      >
        Today
      </span>
      <span
        class="right-0 absolute text-[1.6875rem]"
        data-testid="group-0-elapsed-time"
      >
        0%
      </span>
    </div>
    <div
      class="list"
    >
      <div
        class="w-full overflow-auto max-h-60 lg:max-h-full overflow-x-hidden"
      >
        <ul
          class="text-left min-h-[0.5rem]"
        />
        <button
          class="no-underline m-2 text-base block button-class float-right"
          data-testid="shortTermList-edit-priorities-link"
          style="color: rgb(79, 79, 79);"
        >
          Options
        </button>
        <div
          class="text-center fade-in-1s hidden"
        >
          <form
            class="inline-block w-full mb-2"
          >
            <input
              class="todo leading-none p-2"
              data-testid="todo-input-shortTermList"
              style="color: rgb(79, 79, 79); background-color: rgb(207, 207, 207); border-color: #4f4f4f;"
              type="text"
            />
             
            <input
              class="text-base background-none shadow-none border-none"
              data-testid="shortTermList-add-item-button"
              style="color: rgb(79, 79, 79);"
              type="submit"
              value="Add"
            />
          </form>
          <button
            class="border-none bg-transparent cursor-pointer text-base no-underline link"
            style="color: rgb(79, 79, 79);"
          >
            Clear Done
          </button>
           | 
          <button
            class="border-none bg-transparent cursor-pointer text-base no-underline"
            style="color: rgb(79, 79, 79);"
          >
            Clear All
          </button>
           | 
          <button
            class="border-none bg-transparent cursor-pointer text-base no-underline"
            style="color: rgb(79, 79, 79);"
          >
            Hide
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="fade-in-up-1s align-top m-2 w-full lg:w-3/10 inline-block"
    data-testid="group-container"
  >
    <div
      class="pb-1 text-2xl border-current text-center border-b"
      data-testid="group-1-header"
      role="columnheader"
      tabindex="0"
    >
      <div
        class="absolute left-0"
      >
        <div
          class="inline-block"
          data-testid="group-1-settings"
        >
          <span
            aria-hidden="true"
            class="cursor-pointer mr-2 hidden"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-gears "
              data-icon="gears"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M308.5 135.3c7.1-6.3 9.9-16.2 6.2-25c-2.3-5.3-4.8-10.5-7.6-15.5L304 89.4c-3-5-6.3-9.9-9.8-14.6c-5.7-7.6-15.7-10.1-24.7-7.1l-28.2 9.3c-10.7-8.8-23-16-36.2-20.9L199 27.1c-1.9-9.3-9.1-16.7-18.5-17.8C173.9 8.4 167.2 8 160.4 8h-.7c-6.8 0-13.5 .4-20.1 1.2c-9.4 1.1-16.6 8.6-18.5 17.8L115 56.1c-13.3 5-25.5 12.1-36.2 20.9L50.5 67.8c-9-3-19-.5-24.7 7.1c-3.5 4.7-6.8 9.6-9.9 14.6l-3 5.3c-2.8 5-5.3 10.2-7.6 15.6c-3.7 8.7-.9 18.6 6.2 25l22.2 19.8C32.6 161.9 32 168.9 32 176s.6 14.1 1.7 20.9L11.5 216.7c-7.1 6.3-9.9 16.2-6.2 25c2.3 5.3 4.8 10.5 7.6 15.6l3 5.2c3 5.1 6.3 9.9 9.9 14.6c5.7 7.6 15.7 10.1 24.7 7.1l28.2-9.3c10.7 8.8 23 16 36.2 20.9l6.1 29.1c1.9 9.3 9.1 16.7 18.5 17.8c6.7 .8 13.5 1.2 20.4 1.2s13.7-.4 20.4-1.2c9.4-1.1 16.6-8.6 18.5-17.8l6.1-29.1c13.3-5 25.5-12.1 36.2-20.9l28.2 9.3c9 3 19 .5 24.7-7.1c3.5-4.7 6.8-9.5 9.8-14.6l3.1-5.4c2.8-5 5.3-10.2 7.6-15.5c3.7-8.7 .9-18.6-6.2-25l-22.2-19.8c1.1-6.8 1.7-13.8 1.7-20.9s-.6-14.1-1.7-20.9l22.2-19.8zM112 176a48 48 0 1 1 96 0 48 48 0 1 1 -96 0zM504.7 500.5c6.3 7.1 16.2 9.9 25 6.2c5.3-2.3 10.5-4.8 15.5-7.6l5.4-3.1c5-3 9.9-6.3 14.6-9.8c7.6-5.7 10.1-15.7 7.1-24.7l-9.3-28.2c8.8-10.7 16-23 20.9-36.2l29.1-6.1c9.3-1.9 16.7-9.1 17.8-18.5c.8-6.7 1.2-13.5 1.2-20.4s-.4-13.7-1.2-20.4c-1.1-9.4-8.6-16.6-17.8-18.5L583.9 307c-5-13.3-12.1-25.5-20.9-36.2l9.3-28.2c3-9 .5-19-7.1-24.7c-4.7-3.5-9.6-6.8-14.6-9.9l-5.3-3c-5-2.8-10.2-5.3-15.6-7.6c-8.7-3.7-18.6-.9-25 6.2l-19.8 22.2c-6.8-1.1-13.8-1.7-20.9-1.7s-14.1 .6-20.9 1.7l-19.8-22.2c-6.3-7.1-16.2-9.9-25-6.2c-5.3 2.3-10.5 4.8-15.6 7.6l-5.2 3c-5.1 3-9.9 6.3-14.6 9.9c-7.6 5.7-10.1 15.7-7.1 24.7l9.3 28.2c-8.8 10.7-16 23-20.9 36.2L315.1 313c-9.3 1.9-16.7 9.1-17.8 18.5c-.8 6.7-1.2 13.5-1.2 20.4s.4 13.7 1.2 20.4c1.1 9.4 8.6 16.6 17.8 18.5l29.1 6.1c5 13.3 12.1 25.5 20.9 36.2l-9.3 28.2c-3 9-.5 19 7.1 24.7c4.7 3.5 9.5 6.8 14.6 9.8l5.4 3.1c5 2.8 10.2 5.3 15.5 7.6c8.7 3.7 18.6 .9 25-6.2l19.8-22.2c6.8 1.1 13.8 1.7 20.9 1.7s14.1-.6 20.9-1.7l19.8 22.2zM464 304a48 48 0 1 1 0 96 48 48 0 1 1 0-96z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
        <div
          class="inline-block"
          data-testid="group-1-copy"
        >
          <span
            aria-hidden="true"
            class="cursor-pointer mr-2 hidden"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-copy "
              data-icon="copy"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M208 0H332.1c12.7 0 24.9 5.1 33.9 14.1l67.9 67.9c9 9 14.1 21.2 14.1 33.9V336c0 26.5-21.5 48-48 48H208c-26.5 0-48-21.5-48-48V48c0-26.5 21.5-48 48-48zM48 128h80v64H64V448H256V416h64v48c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V176c0-26.5 21.5-48 48-48z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </div>
      <span
        class="text-[1.6875rem]"
        data-testid="group-1-title"
      >
        Month
      </span>
      <span
        class="right-0 absolute text-[1.6875rem]"
        data-testid="group-1-elapsed-time"
      >
        0%
      </span>
    </div>
    <div
      class="list"
    >
      <div
        class="w-full overflow-auto max-h-60 lg:max-h-full overflow-x-hidden"
      >
        <ul
          class="text-left min-h-[0.5rem]"
        />
        <button
          class="no-underline m-2 text-base block button-class float-right"
          data-testid="mediumTermList-edit-priorities-link"
          style="color: rgb(79, 79, 79);"
        >
          Options
        </button>
        <div
          class="text-center fade-in-1s hidden"
        >
          <form
            class="inline-block w-full mb-2"
          >
            <input
              class="todo leading-none p-2"
              data-testid="todo-input-mediumTermList"
              style="color: rgb(79, 79, 79); background-color: rgb(207, 207, 207); border-color: #4f4f4f;"
              type="text"
            />
             
            <input
              class="text-base background-none shadow-none border-none"
              data-testid="mediumTermList-add-item-button"
              style="color: rgb(79, 79, 79);"
              type="submit"
              value="Add"
            />
          </form>
          <button
            class="border-none bg-transparent cursor-pointer text-base no-underline link"
            style="color: rgb(79, 79, 79);"
          >
            Clear Done
          </button>
           | 
          <button
            class="border-none bg-transparent cursor-pointer text-base no-underline"
            style="color: rgb(79, 79, 79);"
          >
            Clear All
          </button>
           | 
          <button
            class="border-none bg-transparent cursor-pointer text-base no-underline"
            style="color: rgb(79, 79, 79);"
          >
            Hide
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="fade-in-up-1s align-top m-2 w-full lg:w-3/10 inline-block"
    data-testid="group-container"
  >
    <div
      class="pb-1 text-2xl border-current text-center border-b"
      data-testid="group-2-header"
      role="columnheader"
      tabindex="0"
    >
      <div
        class="absolute left-0"
      >
        <div
          class="inline-block"
          data-testid="group-2-settings"
        >
          <span
            aria-hidden="true"
            class="cursor-pointer mr-2 hidden"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-gears "
              data-icon="gears"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M308.5 135.3c7.1-6.3 9.9-16.2 6.2-25c-2.3-5.3-4.8-10.5-7.6-15.5L304 89.4c-3-5-6.3-9.9-9.8-14.6c-5.7-7.6-15.7-10.1-24.7-7.1l-28.2 9.3c-10.7-8.8-23-16-36.2-20.9L199 27.1c-1.9-9.3-9.1-16.7-18.5-17.8C173.9 8.4 167.2 8 160.4 8h-.7c-6.8 0-13.5 .4-20.1 1.2c-9.4 1.1-16.6 8.6-18.5 17.8L115 56.1c-13.3 5-25.5 12.1-36.2 20.9L50.5 67.8c-9-3-19-.5-24.7 7.1c-3.5 4.7-6.8 9.6-9.9 14.6l-3 5.3c-2.8 5-5.3 10.2-7.6 15.6c-3.7 8.7-.9 18.6 6.2 25l22.2 19.8C32.6 161.9 32 168.9 32 176s.6 14.1 1.7 20.9L11.5 216.7c-7.1 6.3-9.9 16.2-6.2 25c2.3 5.3 4.8 10.5 7.6 15.6l3 5.2c3 5.1 6.3 9.9 9.9 14.6c5.7 7.6 15.7 10.1 24.7 7.1l28.2-9.3c10.7 8.8 23 16 36.2 20.9l6.1 29.1c1.9 9.3 9.1 16.7 18.5 17.8c6.7 .8 13.5 1.2 20.4 1.2s13.7-.4 20.4-1.2c9.4-1.1 16.6-8.6 18.5-17.8l6.1-29.1c13.3-5 25.5-12.1 36.2-20.9l28.2 9.3c9 3 19 .5 24.7-7.1c3.5-4.7 6.8-9.5 9.8-14.6l3.1-5.4c2.8-5 5.3-10.2 7.6-15.5c3.7-8.7 .9-18.6-6.2-25l-22.2-19.8c1.1-6.8 1.7-13.8 1.7-20.9s-.6-14.1-1.7-20.9l22.2-19.8zM112 176a48 48 0 1 1 96 0 48 48 0 1 1 -96 0zM504.7 500.5c6.3 7.1 16.2 9.9 25 6.2c5.3-2.3 10.5-4.8 15.5-7.6l5.4-3.1c5-3 9.9-6.3 14.6-9.8c7.6-5.7 10.1-15.7 7.1-24.7l-9.3-28.2c8.8-10.7 16-23 20.9-36.2l29.1-6.1c9.3-1.9 16.7-9.1 17.8-18.5c.8-6.7 1.2-13.5 1.2-20.4s-.4-13.7-1.2-20.4c-1.1-9.4-8.6-16.6-17.8-18.5L583.9 307c-5-13.3-12.1-25.5-20.9-36.2l9.3-28.2c3-9 .5-19-7.1-24.7c-4.7-3.5-9.6-6.8-14.6-9.9l-5.3-3c-5-2.8-10.2-5.3-15.6-7.6c-8.7-3.7-18.6-.9-25 6.2l-19.8 22.2c-6.8-1.1-13.8-1.7-20.9-1.7s-14.1 .6-20.9 1.7l-19.8-22.2c-6.3-7.1-16.2-9.9-25-6.2c-5.3 2.3-10.5 4.8-15.6 7.6l-5.2 3c-5.1 3-9.9 6.3-14.6 9.9c-7.6 5.7-10.1 15.7-7.1 24.7l9.3 28.2c-8.8 10.7-16 23-20.9 36.2L315.1 313c-9.3 1.9-16.7 9.1-17.8 18.5c-.8 6.7-1.2 13.5-1.2 20.4s.4 13.7 1.2 20.4c1.1 9.4 8.6 16.6 17.8 18.5l29.1 6.1c5 13.3 12.1 25.5 20.9 36.2l-9.3 28.2c-3 9-.5 19 7.1 24.7c4.7 3.5 9.5 6.8 14.6 9.8l5.4 3.1c5 2.8 10.2 5.3 15.5 7.6c8.7 3.7 18.6 .9 25-6.2l19.8-22.2c6.8 1.1 13.8 1.7 20.9 1.7s14.1-.6 20.9-1.7l19.8 22.2zM464 304a48 48 0 1 1 0 96 48 48 0 1 1 0-96z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
        <div
          class="inline-block"
          data-testid="group-2-copy"
        >
          <span
            aria-hidden="true"
            class="cursor-pointer mr-2 hidden"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-copy "
              data-icon="copy"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M208 0H332.1c12.7 0 24.9 5.1 33.9 14.1l67.9 67.9c9 9 14.1 21.2 14.1 33.9V336c0 26.5-21.5 48-48 48H208c-26.5 0-48-21.5-48-48V48c0-26.5 21.5-48 48-48zM48 128h80v64H64V448H256V416h64v48c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V176c0-26.5 21.5-48 48-48z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </div>
      <span
        class="text-[1.6875rem]"
        data-testid="group-2-title"
      >
        Year
      </span>
      <span
        class="right-0 absolute text-[1.6875rem]"
        data-testid="group-2-elapsed-time"
      >
        0%
      </span>
    </div>
    <div
      class="list"
    >
      <div
        class="w-full overflow-auto max-h-60 lg:max-h-full overflow-x-hidden"
      >
        <ul
          class="text-left min-h-[0.5rem]"
        />
        <button
          class="no-underline m-2 text-base block button-class float-right"
          data-testid="longTermList-edit-priorities-link"
          style="color: rgb(79, 79, 79);"
        >
          Options
        </button>
        <div
          class="text-center fade-in-1s hidden"
        >
          <form
            class="inline-block w-full mb-2"
          >
            <input
              class="todo leading-none p-2"
              data-testid="todo-input-longTermList"
              style="color: rgb(79, 79, 79); background-color: rgb(207, 207, 207); border-color: #4f4f4f;"
              type="text"
            />
             
            <input
              class="text-base background-none shadow-none border-none"
              data-testid="longTermList-add-item-button"
              style="color: rgb(79, 79, 79);"
              type="submit"
              value="Add"
            />
          </form>
          <button
            class="border-none bg-transparent cursor-pointer text-base no-underline link"
            style="color: rgb(79, 79, 79);"
          >
            Clear Done
          </button>
           | 
          <button
            class="border-none bg-transparent cursor-pointer text-base no-underline"
            style="color: rgb(79, 79, 79);"
          >
            Clear All
          </button>
           | 
          <button
            class="border-none bg-transparent cursor-pointer text-base no-underline"
            style="color: rgb(79, 79, 79);"
          >
            Hide
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
